{{ cookiecutter.project_slug }}/.github/workflows/docs.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
```

Comment:
I can see arguments for and against. If your docs include a user guide, you might only want that to be built on release, as users should really only be installing releases anyway, so you want the guide to match up with the last release. On the other hand, for anything developer-related (e.g. API documentation), you probably want it to be up to date with `main` because there's not really any point in instructions that won't work with the latest code!

How about leaving it as is, but adding a comment below explaining how to run it on release only?